src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyFlowGraph.cs

```diff
@@ -0,0 +1,197 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Maestro.Client.Models;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.DarcLib
+{
+    /// <summary>
+    ///     This graph build
+    /// </summary>
+    public class DependencyFlowGraph
+    {
+        public DependencyFlowGraph(List<DependencyFlowNode> nodes, List<DependencyFlowEdge> edges)
+        {
+            Nodes = nodes;
+            Edges = edges;
+        }
+
+        public List<DependencyFlowNode> Nodes { get; set; }
+        public List<DependencyFlowEdge> Edges { get; set; }
+
+        public void RemoveNode(DependencyFlowNode node)
+        {
+            // Remove the node from the list, then remove corresponding edges
+            if (Nodes.Remove(node))
+            {
+                foreach (DependencyFlowEdge incomingEdge in node.IncomingEdges)
+                {
+                    // Don't use RemoveEdge as it assumes that this node will remain in the graph and
+                    // will cause modification of this IncomingEdges collection while iterating
+                    incomingEdge.From.OutgoingEdges.Remove(incomingEdge);
+                }
+                foreach (DependencyFlowEdge outgoingEdge in node.OutgoingEdges)
+                {
+                    // Don't use RemoveEdge as it assumes that this node will remain in the graph and
+                    // will cause modification of this OutgoingEdges collection while iterating
+                    DependencyFlowNode targetNode = outgoingEdge.To;
+                    outgoingEdge.To.IncomingEdges.Remove(outgoingEdge);
+                    // Recalculate the input channels
+                    targetNode.InputChannels = new HashSet<string>(targetNode.IncomingEdges.Select(e => e.Subscription.Channel.Name));
```

Comment:
This is duplicated, it should probably be in a function somwhere. Should this ignore case?